internal/oci/runtime_oci.go

```diff
@@ -591,21 +594,25 @@ func waitContainerStop(ctx context.Context, c *Container, timeout time.Duration,
 		}
 	}
 
-	c.state.Finished = time.Now()
 	return nil
 }
 
 // StopContainer stops a container. Timeout is given in seconds.
-func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout int64) error {
+func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout int64) (err error) {
 	c.opLock.Lock()
-	defer c.opLock.Unlock()
 
 	// Check if the process is around before sending a signal
 	process, err := findprocess.FindProcess(c.state.Pid)
 	if err == findprocess.ErrNotFound {
-		c.state.Finished = time.Now()
+		c.opLock.Unlock()
+		c.SetFinished(time.Now())
 		return nil
 	}
+	defer func() {
+		if err != nil {
```

Comment:
maybe errRet to stick with the other functions?

Reply:
Addressed in next push.